test/link/macho/needed_library/build.zig

```diff
@@ -31,6 +31,7 @@ pub fn build(b: *Builder) void {
     const check = exe.checkObject(.macho);
     check.checkStart("cmd LOAD_DYLIB");
     check.checkNext("name @rpath/liba.dylib");
+    test_step.dependOn(&check.step);
```

Comment:
`runAndCompare()` down below already depends on the `check` step. The build steps are a graph, meaning that this statement is unneeded, as the `test_step` already depends on the `run_cmd` step.

Reply:
I thought this initially too, but empirically it does not work this way. Maybe this is just a bug somewhere else, but some of these tests started failing when I added these `dependOn`, and intentionally breaking a check did not cause any test failures without them.

Reply:
Do you happen to remember which cases this was occurring at? `dylib` and `dead_strip` did indeed miss the `dependOn`'s but the other ones *should* be working. I'll happily take a look at why this wasn't working, but it would save me some time if this was happening to all of them, or just those cases.

Reply:
Previously there was no way for any test that calls `checkInSymtab` to pass, so those are precisely the tests that I know for sure were not actually running.  It's certainly possible that I misdiagnosed the issue and added more than were necessary, given that I am totally lost as to the difference between the `dependOn` that was added to `needed_library` and the one that was added to `dylib`, since in both cases `runAndCompare` is called and added as a dependency to the top level test step.

I should also mention that I was a bit confused as to why `runAndCompare` is a sub-step of `checkObject` in the first place.  If I am understanding correctly, it seems like something that you might want to do on any executable, regardless of whether you also want to create a `checkObject` step.

I would certainly appreciate someone who understands the build system a bit better taking a look at this.  Also, I am just now noticing that the `dependOn` that I added to `dylib` is slightly misplaced, so if there is a follow-up PR maybe it could fix that too. :eyes:

Reply:
`runAndCompare` is a simple wrapping function part of `checkObject` that creates an `EmulatableStep` because we use it very often in our linker tests. The same step can be run on any executable (see `runEmulatable` in lib/std/build.zig). I'll take a look into the step to see why it isn't calling any of the dependable steps to prevent any future regressions. Thank you for providing the information.

Reply:
Resolved in #13218.